core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -98,14 +109,13 @@ protected String name() {
     Preconditions.checkArgument(namespace.levels().length >= 1,
         "Missing database in table identifier: %s", namespace);
 
-    Joiner slash = Joiner.on("/");
-    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
     FileSystem fs = Util.getFs(nsPath, conf);
     Set<TableIdentifier> tblIdents = Sets.newHashSet();
 
     try {
       if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
-        throw new NotFoundException("Unknown namespace " + namespace);
+        throw new NoSuchNamespaceException("namespace does not exist: " + namespace.toString());
```

Comment:
Error messages should use sentence case. Please capitalize `Namespace` here.

Also, omit `toString` when it is automatically called, as with string concatenation.